internal/payloads/error_payload.go

```diff
@@ -194,6 +194,11 @@ func NewStatusError(ctx context.Context, message string, err error) *ResponseErr
 }
 
 func NewAWSError(ctx context.Context, message string, err error) *ResponseError {
+	var awsOpErr *smithy.OperationError
+	if errors.As(err, &awsOpErr) && awsOpErr.OperationName == "AssumeRole" {
+		message = fmt.Sprintf("AWS assume role failed: %s", message)
+		return NewResponseError(ctx, http.StatusForbidden, message, err)
+	}
```

Comment:
This does not look correct, operation name does not say anything about resulting HTTP code. Isn’t better to search for `GenericAPIError` code "AccessDenied"?

Reply:
I was actually talking about the middle layer error (ResponseError), you're right, GenericAPIError makes more sense here, fixed! :+1: